### PR TITLE
Add primary key

### LINKED ISSUE #286: Add primary key to oc_e2e_encryption_lock table
As of https://github.com/nextcloud/server/pull/31513 we need to have a primary key.

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Fix #286
Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: lib/Migration/Version1101Date20220316143119.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		if ($schema->hasTable('e2e_encryption_lock')) {
```
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply lib/Migration/Version1101Date20220316143119.php]: Thanks :)

File: lib/Migration/Version1101Date20220316143119.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?php
+
```
Comment (CarlSchwan) [inline lib/Migration/Version1101Date20220316143119.php:2]: copyright :)

Comment (tobiasKaminsky) [reply lib/Migration/Version1101Date20220316143119.php:2]: Done :)

**Comment by tobiasKaminsky** [comment]
I pushed latest copryright change. Seems GH is a bit slow currently.

File: lib/Migration/Version1101Date20220316143119.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+namespace OCA\EndToEndEncryption\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
+ */
```
Comment (nickvergessen) [inline lib/Migration/Version1101Date20220316143119.php:33]:
```suggestion
```

**Review by CarlSchwan — APPROVED** [review]

**Pull request merged**